nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/errors/ProcessingEarlyExit.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.nosqlbench.nb.api.errors;
+
+public class ProcessingEarlyExit extends RuntimeException {
+    private final String message;
+    private final int exitCode;
+
+    public ProcessingEarlyExit(String message, int exitCode) {
+        this.message = message;
+        this.exitCode = exitCode;
+    }
+
+    public String toString() {
+        final String helpmsg = """
+                'ARG'.
+                """
+            .replaceAll("ARG", this.message);
```

Comment:
Consider using STR templates for this kind of message.

Reply:
I should redo this as I was copying a pattern that no longer makes sense.